ocelli-rxnetty/src/main/java/netflix/ocelli/rxnetty/FailureListener.java

```diff
@@ -0,0 +1,38 @@
+package netflix.ocelli.rxnetty;
+
+import rx.Scheduler;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A contract for taking actions upon detecting an unhealthy host. A failure listener instance is always associated with
+ * a unique host, so any action taken on this listener will directly be applied to the associated host.
+ */
+public interface FailureListener {
```

Comment:
ocelli-core has the concept of an InstanceEventListener used to notify the quarantier of ExecutionSuccess and ExecutionFailed events for a host.  These events feed both into metrics used to track performance as well as a quarantining strategy that will make the determination as to whether to quaratine the host and for how long.  Note that the quarantining strategy should be outside of the scope of the client implementation.  With this FailureListener contract here the HttpClientEventsListener must now know the quarantine strategy which does not provide a good separation of concerns.